### PR TITLE
ci: action for updating engines via workflow dispatch

### PR DESCRIPTION
**Pull request by pantharshit00**
This PR adds a new github action which updates the engines via the input received from a workflow dispatch. It also ensures that the tag provided is available on npm.

Example PR: https://github.com/pantharshit00/prisma-1/pull/1
Example Run: https://github.com/pantharshit00/prisma-1/actions/runs/935462151

TODO:
 - [x] Test run on a fork
 - [x] Action for automatically merging PRs when all tests pass starting with a specific title which will be used by the first action
 - [x] ~Acquire secrets for Prisma Bot so that commit is always made by the bot account~. GitHub actions bot is fine.

### CODE REVIEW

File: .github/workflows/update-engines-version.yml
Comment (Jolg42) [inline .github/workflows/update-engines-version.yml:23]:
```suggestion
      - name: Check if tag is available on npm
```
nitpick

File: .github/workflows/update-engines-version.yml
Comment (Jolg42) [inline .github/workflows/update-engines-version.yml:26]: `enr` is a bit cryptic, what is this?

Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml:26]: Its abbreviation for `@prisma/ensure-npm-release` cli

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:5]: TODO: change to `version`

File: .github/workflows/update-engines-version.yml
Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml:26]: https://github.com/prisma/ensure-npm-release

**Comment by Jolg42** [comment]
Looks great!

I think we could improve the PR description, the line return is a bit weird: 
Screen Shot 2021-06-14 at 13 47 29

We could add links to npm packages maybe.

**Comment by pantharshit00** [comment]
Updated description with links:
https://github.com/pantharshit00/prisma-1/pull/1#issue-669447799
image

File: .github/workflows/update-engines-version.yml
Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml]: Fixed

**Comment by Jolg42** [comment]
Should we merge this?

If we do I think we'll have to tweak Renovate to ignore these 2 packages to be updated.

**Comment by pantharshit00** [comment]
https://github.com/pantharshit00/prisma-1/pull/3

Auto merge working demo. I will wait for @janpio ’s review before merging

**Review by Jolg42 — APPROVED** [review]

**Comment by janpio** [comment]
> If we do I think we'll have to tweak Renovate to ignore these 2 packages to be updated.

Optimally this script would be faster, and Renovate would not even open a PR - or close it again automatically when this did its job, right @Jolg42? Then we could keep Renovate unchanged as backup for now and only exclude the package after this is working 💯

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:28]: These commands are blocking until they succeed?

**Comment by pantharshit00** [comment]
Yes, indeed. Renovate will even auto close PRs it makes if package updates in master so we can definitely keep that unchanged.

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:31]:
```suggestion
      - name: Update the engines dependencies
```

File: .github/workflows/update-engines-version.yml
Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml:28]: Yes

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:33]:
```suggestion
          echo 'Updating @prisma/engines and @prisma/engines-version to ${{ github.event.inputs.version }} using pnpm'
```

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:26]:
```suggestion
          echo 'Checking @prisma/engines@${{ github.event.inputs.version }} (and @prisma/engines-version)'
```

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:42]: What happens if we already have a PR open? 

I think this might need to include the inputs.version as well.

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:43]: might not be necessary as we have that configured on the repo level

File: .github/workflows/update-engines-version.yml
Comment (janpio) [inline .github/workflows/update-engines-version.yml:65]:
```suggestion
      - name: Auto approve Pull Request (to trigger Automergeif tests passing)
```

File: .github/workflows/update-engines-version.yml
Comment (janpio) [reply .github/workflows/update-engines-version.yml:28]: How long do they wait? Endless / until the Action times out?

**Review by janpio — APPROVED** [review]
Nice.

Left a small change suggestions, after those this should be 💯 (it it actually works after merge 😆)

**Comment by Jolg42** [comment]
Sure @janpio let's keep renovate unchanged and see what happens!

File: .github/workflows/update-engines-version.yml
Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml:28]: I use exponential backoff so maximum is 30 minutes. If actions times out before that then its another case.

File: .github/workflows/update-engines-version.yml
Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml:42]: It matches the renovate behaviour and updates the PR titles and force pushes the removes the earlier commit.

File: .github/workflows/update-engines-version.yml
Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml:43]: It uses the same github thing

File: .github/workflows/update-engines-version.yml
Comment (pantharshit00) [reply .github/workflows/update-engines-version.yml:42]: Example: https://github.com/pantharshit00/prisma-1/pull/1

**Comment by Jolg42** [comment]
Can't wait to see it working 💯 

Well done @pantharshit00 !

**Pull request merged**
